controllers/feeds/reactionList.js

```diff
@@ -20,9 +22,30 @@ module.exports = async (req, res) => {
       return {...commentRes, latest_children: {comment: sortComment}};
     });
 
+    // get comment users
+    let actors = [];
+    sortByDate.map((comment_lvl1) => {
+      if (!actors.includes(comment_lvl1?.user?.id)) {
+        actors.push(comment_lvl1?.user?.id);
+      }
+      comment_lvl1?.latest_children?.comment?.map((comment_lvl2) => {
+        if (!actors.includes(comment_lvl2?.user?.id)) {
+          actors.push(comment_lvl2?.user?.id);
+        }
+      });
+    });
+    const karmaScores = await UsersFunction.getUsersKarmaScore(User, actors);
+
     const removeSensitiveData = await Promise.all(
       sortByDate.map(async (data) => {
-        return handleAnonymousData(data, req, post.actor.id, myAnonymousId, data.user_id);
+        return handleAnonymousData(
+          data,
+          req,
+          post.actor.id,
+          myAnonymousId,
+          data.user_id,
+          karmaScores
+        );
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The logic for collecting user IDs (`actors`) from the comments seems to be duplicated for both level 1 and level 2 comments. This could be abstracted into a helper function to reduce code duplication and improve readability.

Here's a suggestion on how you can refactor this:

```diff
-    sortByDate.map((comment_lvl1) => {
-      if (!actors.includes(comment_lvl1?.user?.id)) {
-        actors.push(comment_lvl1?.user?.id);
-      }
-      comment_lvl1?.latest_children?.comment?.map((comment_lvl2) => {
-        if (!actors.includes(comment_lvl2?.user?.id)) {
-          actors.push(comment_lvl2?.user?.id);
-        }
-      });
-    });

+    const collectUserIds = (comment) => {
+      if (comment?.user?.id && !actors.includes(comment.user.id)) {
+        actors.push(comment.user.id);
+      }
+      comment?.latest_children?.comment?.forEach(collectUserIds);
+    };
+
+    sortByDate.forEach(collectUserIds);
```

This way, you have a single function `collectUserIds` that handles the collection of user IDs from any level of comments. It's more maintainable and easier to understand.


<!-- This is an auto-generated comment by OSS CodeRabbit -->

Reply:
Sudah disesuaikan